crates/interpreter/src/value.rs

```diff
@@ -71,3 +71,28 @@ pub fn strict_equality(x: &JSValue, y: &JSValue) -> bool {
         _ => false,
     }
 }
+
+pub fn equality(x: &JSValue, y: &JSValue) -> bool {
+    match (x, y) {
```

Comment:
can you add reference to https://tc39.es/ecma262/#sec-abstract-equality-comparison ?
and possibly comment for each step.